### DISCUSSION TITLE
Store permanently colour schema

### DISCUSSION THREAD

**Discussion by TeoTN** [home-assistant/frontend#18061] [Configuration]
I'd like to propose permanent storing of colour schema for the dashboard.
They way it's stored currently is fragile, and every now and then my colours of preference are lost.
The preference is also not shared across devices.
I'd love to be able to pick the primary and secondary colours, have them stored on the device, rather than in the browser, and shared across all my devices!

**Comment by ildar170975** [comment]
What you call a "color schema" is a theme.
There is a default theme (light & dark), and custom themes may be created.
Each device may have its own theme - adapted to screen dimensions & capabilities of these devices (bigger/smaller, color/black-and-white, ...), location of these devices (sunny outdoor, dark cellar, ...).
And it should be this way, imho.

**Reply by TeoTN** [reply]
I don't think one excludes the other. If themes were stored, it could be made in a way that you can pick whichever you want on each device. I just don't want to be forced to remember my favourite HEX colours, and don't want them to be susceptible to random deleting, I don't mind having an option to create more than one

**Reply by ildar170975** [reply]
With a default theme a user may only change these colors:
![image](https://github.com/home-assistant/frontend/assets/71872483/cfa8c223-3bdc-42da-a06a-d3834352da54)
Each color has a corresponding variable (like `-some-themeable-color`).
If you need to manage only two these colors - you may create a custom theme with these 2 vars only:
```
my_theme:
  modes:
    light:
      primary-color: red
      accent-color: green
    dark:
      primary-color: red
      accent-color: green
```
Then you will need to select this custom theme on each device (every time when you reset your local settings, for instance - so the theme is also reset to default).
There is no need to keep in mind these colors anymore.

P.S. A user-selected theme is reset to a default one when a user logout, unfortunately. But this is another issue...
